### PR TITLE
Mob 3650

### PR DESCRIPTION
**Pull request by dianakarh**
## JIRA
[MOB-3650](https://karhoo.atlassian.net/browse/MOB-3650)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [ ] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

**Review by mionescu95 — APPROVED** [review]

File: Podfile.lock [hunk 1]
```diff
@@ -68,7 +68,7 @@ DEPENDENCIES:
   - Braintree/PaymentFlow (~> 4.37)
   - BraintreeDropIn (~> 8.1)
   - FloatingPanel (= 2.0.1)
-  - "KarhooSDK (from `git@github.com:karhoo/karhoo-ios-sdk.git`, branch `develop`)"
+  - "KarhooSDK (from `git@github.com:karhoo/karhoo-ios-sdk.git`, branch `MOB-3649`)"
```
Comment (mohadian) [inline Podfile.lock:71]: I think you can revert back to `develop`

File: Podfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ def common_pods
   pod 'Adyen', '3.7.0'
 #  pod 'KarhooSDK', '1.5.3'
   #pod 'KarhooSDK', :path => '../karhoo-ios-sdk'
-  pod 'KarhooSDK', :git => 'git@github.com:karhoo/karhoo-ios-sdk.git', :branch => 'develop'
+  pod 'KarhooSDK', :git => 'git@github.com:karhoo/karhoo-ios-sdk.git', :branch => 'MOB-3649'
```
Comment (mohadian) [inline Podfile:17]: please revert back to `develop`

File: KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift [hunk 3]
```diff
@@ -202,41 +300,58 @@ final class KarhooLoyaltyPresenter: LoyaltyPresenter {
...
-            view?.updateLoyaltyFeatures(showEarnRelatedUI: true, showBurnRelatedUI: viewModel?.canBurn ?? false)
-            showcaseBurnModeType = .valid
+    private func handleEarnPointsCallError() {
+        // TODO: Once the slug is added to the error returned from the server, uncomment and update the snippet below
+        // to show the unsupported currency error only when it is indeed the case
+        // and delete the call to updateUIWithError on the next line
+        if currentMode == .earn, (viewModel?.canEarn ?? false) {
+            updateUIWithError(message: UITexts.Errors.unsupportedCurrency)
         }
+//                switch error?.type {
```
Comment (mohadian) [inline KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift:333]: Do we need to keep the commented code?

File: KarhooUISDK/Translations/Base.lproj/Localizable.strings [hunk 4]
```diff
@@ -236,7 +236,7 @@
 "Text.Booking.PrebookConfirmed.RideDetails" = "Ride Details";
 
 /* Booking */
-"Text.Booking.RequestCar" = "Book";
+"Text.Booking.RequestCar" = "PAY AND BOOK";
```
Comment (mohadian) [inline KarhooUISDK/Translations/Base.lproj/Localizable.strings:239]: I am wondering if this is aligned with Android? Do we need to do it as part of this PR?

Check
https://github.com/karhoo/karhoo-android-ui-sdk/pull/256/files

File: Podfile.lock (hunk 1 above)
Comment (dianakarh) [reply Podfile.lock]: done

File: Podfile (hunk 2 above)
Comment (dianakarh) [reply Podfile]: done

File: KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift (hunk 3 above)
Comment (dianakarh) [reply KarhooUISDK/UIComponents/LoyaltyView/KarhooLoyaltyPresenter.swift]: it's part of a TODO for when the slug issue will be resolved (hopefully soon)

**Review by mohadian — APPROVED** [review]
Approved, 
@EdwardW2 please review the below comments with Olivier next week
https://github.com/karhoo/karhoo-ios-ui-sdk/pull/221/files#r773212290

**Review by mohadian — APPROVED** [review]

File: KarhooUISDK/Translations/Base.lproj/Localizable.strings (hunk 4 above)
Comment (EdwardW2) [reply KarhooUISDK/Translations/Base.lproj/Localizable.strings:239]: @dianakarh Yes, i do believe this string was updated on Android. I do remember updating the book button not too long ago

Comment (mohadian) [reply KarhooUISDK/Translations/Base.lproj/Localizable.strings:239]: Can you please double-check the A/Cs and make sure both platforms are aligned? It seems Android is different and this discrepancy does not make sense.

https://github.com/karhoo/karhoo-android-ui-sdk/blob/develop/uisdk/src/main/res/values/strings.xml#L33

**Pull request merged**